src/main/webapp/script.js

```diff
@@ -41,47 +42,69 @@ function initCommentSlider() {
  * @return {JSON} comments json object of comment objects
  */
 async function getYouTubeComments(url) {
-  url = extractYouTubeUrl(url);
-  const numComments = document.getElementById(SLIDER_NAME).value;
-  const response = await fetch('/YouTubeComments?url=' + url +
-    '&numComments=' + numComments);
-  const comments = await response.json();
-  return comments;
+  try {
+    url = extractYouTubeUrl(url);
+    const numComments = document.getElementById(SLIDER_NAME).value;
+    const response = await fetch('/YouTubeComments?url=' + url +
+      '&numComments=' + numComments);
+    const comments = await response.json();
+    return comments;
+  } catch (err) {
+    throw new Error('Error retrieving comments.');
+  }
 }
 
 /**
  * Wrapper function for preparing onClick function
  */
 function onButtonPress() {
   $('#submit-link-btn').click(function() {
-    showLoadingGif();  
-    const urlInput = document.getElementById('link-input').value;
-    updateUIWithVideoContext(urlInput);
-    displayOverallResults(urlInput);
+    try {
+      toggleErrorOutput('none');
+      showLoadingGif();
+      const urlInput = document.getElementById('link-input').value;
+      updateUIWithVideoContext(urlInput);
+      displayOverallResults(urlInput);
+    } catch (err) {
+      displayError(err.message);
+    }
   });
 }
 
+// Toggles visibility of error message
+function toggleErrorOutput(mode) {
+  document.getElementById(ERROR_OUTPUT_ID).style.display = mode;
+}
+
+function displayError(message) {
+  hideLoadingGif();
+  toggleErrorOutput('block');
+  document.getElementById('error-details').innerText = message;
+}
```

Comment:
We probably also want this function to clear the other page elements.

Right now if an error is encountered part way through executing a request (i.e. perhaps after it has retreieved the video context but before the sentiment analysis has been done), then the rest of the page will still be in a "partial" state.

I think the easiest way to handle this would be to have a container that could be populated with video results in normal execution but would be populated with a formatted error message in error scenarios.

Reply:
Alright, I feel like this may merit waiting until everything is merged then making another small pr just because I think my podmate's pr's are changing said other page elements.

Reply:
Yeah we are changing the containers that get used for the result objects. 

But as long as we don't break this new error logic when we make those changes, I think it's fine to merge this PR (with some minor revisions and Cis's LGMT)

Reply:
This is fine with me as well.